### PR TITLE
test: remove CreateArray, refactor Append

### PR DESCRIPTION
**Pull request by bcollamore**

### CODE REVIEW

File: Philips.CodeAnalysis.Test/Maintainability/Maintainability/DontLockNewObjectAnalyzerTest.cs [hunk 1]
```diff
@@ -1,11 +1,13 @@
 ﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
 using System;
+using System.Reflection.Metadata;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Philips.CodeAnalysis.Common;
 using Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability;
 using Philips.CodeAnalysis.Test.Helpers;
 using Philips.CodeAnalysis.Test.Verifiers;
+using static System.Runtime.InteropServices.JavaScript.JSType;
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/Maintainability/Maintainability/DontLockNewObjectAnalyzerTest.cs:10]: Please remove this unused using

File: Philips.CodeAnalysis.Test/MsTest/DataTestMethodsHaveDataRowsAnalyzerTest.cs [hunk 2]
```diff
@@ -79,7 +79,7 @@ public class Tests
 		[TestCategory(TestDefinitions.UnitTests)]
 		public void DataTestMethodsMustHaveDataRows2(string arg, bool isError)
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/MsTest/DataTestMethodsHaveDataRowsAnalyzerTest.cs:80]: Let's split this method also,these isError parameters are confusing

File: Philips.CodeAnalysis.Test/MsTest/DataTestMethodsHaveDataRowsAnalyzerTest.cs [hunk 3]
```diff
@@ -106,7 +108,7 @@ public class Tests
 		[TestCategory(TestDefinitions.UnitTests)]
 		public void TestMethodsMustNotHaveDataRows(string arg, bool isError)
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/MsTest/DataTestMethodsHaveDataRowsAnalyzerTest.cs:109]: Same here

File: Philips.CodeAnalysis.Test/MsTest/TestMethodsShouldHaveValidReturnTypesAnalyzerTest.cs [hunk 4]
```diff
@@ -51,7 +51,15 @@ public class Tests
 	public {(isAsync ? "async" : string.Empty)} {returnType} Foo() {{ throw new Exception(); }}
 }}";
 
-			VerifyDiagnostic(code, isError ? DiagnosticResultHelper.CreateArray(DiagnosticIds.TestMethodsMustHaveValidReturnType) : Array.Empty<DiagnosticResult>());
+			if (isError)
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/MsTest/TestMethodsShouldHaveValidReturnTypesAnalyzerTest.cs:54]: And here

**Review by ynsehoornenborg — APPROVED** [review]

File: Philips.CodeAnalysis.Test/Maintainability/Maintainability/DontLockNewObjectAnalyzerTest.cs (hunk 1 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/Maintainability/Maintainability/DontLockNewObjectAnalyzerTest.cs:10]: nice catch. will take care of in round 2.

File: Philips.CodeAnalysis.Test/MsTest/DataTestMethodsHaveDataRowsAnalyzerTest.cs (hunk 2 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/MsTest/DataTestMethodsHaveDataRowsAnalyzerTest.cs:80]: Round 2

**Pull request merged**